src/conn.c

```diff
@@ -127,7 +134,8 @@ static void raft_connect(struct conn *c)
 	/* Close the connection without actually closing the transport, since
 	 * the stream will be used by raft */
```

Comment:
This comment is a little bit outdated now that we call `conn_transport_close_cb`. Either the function name is misleading (but it is in fact a callback for closing the transport) or the comment is a little bit short now and should say something but we still free the resources.

However, this leads me to believe you can call `conn_transport_close_cb` without closing the transport which means that maybe it is clearer if we changed the name of the function and we didn't use it as the callback for uv_close(transport, ...).

Reply:
Indeed, `conn_transport_close_cb` is not responsible for closing the transport, rather it's the callback that runs after the transport has been closed. This naming convention is confusing but it's also used a lot in the existing code. In this case I like the idea of renaming that function to make more clear what it does, but I think we should still re-use it as the callback passed to `uv_close(transport)` because the two uses are conceptually connected. I can update the comment to explain this better as well.